### DISCUSSION TITLE
Concatenate arrays to add timesteps

### DISCUSSION THREAD

**Discussion by hug-fr** [pydata/xarray#6608] [Q&A] [answered]
Let's say I have two datasets, both with the same two dimensions and the same n variables. One dimension is time and the other is gid. 

Some variables depend on both but some others depend only on gid (for instance lat/lon). Between the tow datasets the gid coordinates are the same but the time coordinates are complementary. 

What I would like is simply join these two datasets in order to have just one with all the time steps. I tried
new_ds = xr.concat ([ds1, ds2], dim = 'time')

But the result is not the expected one. The variables depending only on the gid dim now depend on both time and gid dimension. 

For sur, I don't understand something and any help would be greatly appreciated

**Comment by mathause** [comment]
Can you try `xr.combine_by_coords` or `xr.combine_nested` - see also the docs: https://docs.xarray.dev/en/stable/user-guide/combining.html#combining-along-multiple-dimensions

**Comment by keewis** [comment, marked as answer]
the `coords` and `data_vars` parameters of `concat` allow controlling what is concatenated. Usually, setting them to `"minimal"` should be enough:
```python
new_ds = xr.concat([ds1, ds2], dim="time", data_vars="minimal", coords="minimal")
```

**Reply by hug-fr** [reply]
Worked like a charm! Thank you very much

**Comment by hug-fr** [comment]
Thank you very much for your answers. I tried all the suggested methods and here are the results.

**Dataset 1**
```
Dimensions:     (time: 138, gid: 1047)
Coordinates:
  * time        (time) datetime64[ns] 1982-08-01T06:00:00 ... 1982-08-06T23:0...
  * gid         (gid) int64 5363 5364 5365 5444 5445 ... 9057 9058 9059 9060
Data variables: (12/14)
    lat         (gid) float64 ...
    lon         (gid) float64 ...
```
**Dataset 2**
```
Dimensions:     (time: 6, gid: 1047)
Coordinates:
  * time        (time) datetime64[ns] 1982-08-01 ... 1982-08-01T05:00:00
  * gid         (gid) int64 5363 5364 5365 5444 5445 ... 9057 9058 9059 9060
Data variables: (12/14)
    lat         (gid) float64 ...
    lon         (gid) float64 ...
```
**Here is the result with combine_by_coords: did not provide the result I'm looking for**
```
Dimensions:     (time: 144, gid: 1047)
Coordinates:
  * time        (time) datetime64[ns] 1982-08-01 ... 1982-08-06T23:00:00
  * gid         (gid) int64 5363 5364 5365 5444 5445 ... 9057 9058 9059 9060
Data variables: (12/14)
    lat         (time, gid) float64 46.48 46.47 46.47 ... 43.42 43.41 43.41
    lon         (time, gid) float64 7.498 7.602 7.706 ... 6.531 6.629 6.728
```
**Here is the result with combine_nested with concat_dim = 'time': Uh! I don't really understand how the output happens with 6 additional time steps and all the dims applied to my lon/lat variables**
```
Dimensions:     (time: 150, gid: 1047)
Coordinates:
  * time        (time) datetime64[ns] 1982-08-01 ... 1982-08-06T23:00:00
  * gid         (gid) int64 5363 5364 5365 5444 5445 ... 9057 9058 9059 9060
Data variables: (12/14)
    lat         (time, gid) float64 46.48 46.47 46.47 ... 43.42 43.41 43.41
    lon         (time, gid) float64 7.498 7.602 7.706 ... 6.531 6.629 6.728
```
**Here is the result with concat using the "minimal" setting for both parameters as suggested: it looks like I got the expected result! Wouhou!**
```
Dimensions:     (time: 144, gid: 1047)
Coordinates:
  * time        (time) datetime64[ns] 1982-08-01 ... 1982-08-06T23:00:00
  * gid         (gid) int64 5363 5364 5365 5444 5445 ... 9057 9058 9059 9060
Data variables: (12/14)
    lat         (gid) float64 46.48 46.47 46.47 46.48 ... 43.42 43.41 43.41
    lon         (gid) float64 7.498 7.602 7.706 5.411 ... 6.531 6.629 6.728
```
